### PR TITLE
SwiftDriverTests: adjust for Windows

### PR DESCRIPTION
**Pull request by compnerd**
This adjusts the tests for Windows.  With this set of changes, the
ExplicitModuleBuildTests all pass on Windows (except those which use a
macOS target triple).

### CODE REVIEW

File: Tests/SwiftDriverTests/ExplicitModuleBuildTests.swift [hunk 1]
```diff
@@ -750,7 +770,7 @@ final class ExplicitModuleBuildTests: XCTestCase {
         $0 <<< "import G;"
       }
       let packageRootPath = URL(fileURLWithPath: #file).pathComponents
-        .prefix(while: { $0 != "Tests" }).joined(separator: "/").dropFirst()
+        .prefix(while: { $0 != "Tests" }).joined(separator: "/")
```
Comment (compnerd) [inline Tests/SwiftDriverTests/ExplicitModuleBuildTests.swift:773]: Not sure how to handle this.  The problem is that the first component is the drive letter unlike Unix paths where `/` is the first component.

**Comment by compnerd** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/ExplicitModuleBuildTests.swift [hunk 2]
```diff
@@ -585,22 +599,24 @@ final class ExplicitModuleBuildTests: XCTestCase {
...
       }
 
-      let packageRootPath = URL(fileURLWithPath: #file).pathComponents
-          .prefix(while: { $0 != "Tests" }).joined(separator: "/").dropFirst()
-      let testInputsPath = packageRootPath + "/TestInputs"
-      let cHeadersPath : String = testInputsPath + "/ExplicitModuleBuilds/CHeaders"
-      let swiftModuleInterfacesPath : String = testInputsPath + "/ExplicitModuleBuilds/Swift"
+      let cHeadersPath: String =
+          testInputsPath.appending(component: "ExplicitModuleBuilds")
+                        .appending(compnent: "CHeaders")
```
Comment (artemcm) [inline Tests/SwiftDriverTests/ExplicitModuleBuildTests.swift:604]:
```suggestion
                        .appending(component: "CHeaders")
```

**Comment by compnerd** [comment]
Thanks; it took a while, but finally saw how this could be put together so that it was much cleaner.

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**